data_sources/datastore_data.py

```diff
@@ -9,6 +9,55 @@
 from models.call_history_model import CallHistory
 
 
+class DataStoreClient:
```

Comment:
Might be a pain with how the rest of the repo is written, but I would be tempted to have this be a CallHistoryClient, with all of the related methods on it. Generally trying to think of the intent of the class, which is to manage call history reports, the fact that datastore is used is just an implementation detail.

Although I don't want to trigger another massive refactor.